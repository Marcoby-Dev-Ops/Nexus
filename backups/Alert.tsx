import React from 'react';
import { cn } from '@/lib/utils';
import PropTypes from 'prop-types';

/**
 * @interface AlertProps
 * @description Props for the Alert component.
 */
export interface AlertProps extends React.HTMLAttributes<HTMLDivElement> {
  variant?: 'default' | 'error' | 'success' | 'warning';
  action?: React.ReactNode;
}

/**
 * @name Alert
 * @description A notification component for feedback messages.
 * @param {AlertProps} props - The props for the component.
 * @returns {JSX.Element} The rendered Alert component.
 */
export const Alert: React.FC<AlertProps> = ({
  children,
  className,
  variant = 'default',
  action,
  ...props
}) => {
  const variants = {
    default: 'bg-muted text-foreground',
    error: 'bg-destructive/10 text-destructive',
    success: 'bg-success/10 text-success',
    warning: 'bg-warning/10 text-warning'
  };

  return (
    <div
      className={cn(
        'rounded-lg p-4 flex items-start gap-4',
        variants[variant],
        className
      )}
      {...props}
    >
      <div className="flex-1">{children}</div>
      {action && <div className="flex-shrink-0">{action}</div>}
    </div>
  );
};

Alert.displayName = 'Alert';

Alert.propTypes = {
  variant: PropTypes.oneOf(['default', 'error', 'success', 'warning']),
  action: PropTypes.node,
  className: PropTypes.string,
};

export default Alert; 