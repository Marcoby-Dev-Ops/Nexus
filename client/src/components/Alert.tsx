import React from 'react';
import PropTypes from 'prop-types';

/**
 * @interface AlertProps
 * @description Props for the Alert component.
 */
export interface AlertProps {
  type?: 'info' | 'success' | 'warning' | 'error';
  message: string;
  className?: string;
}

/**
 * @name Alert
 * @description A notification component for feedback messages.
 * @param {AlertProps} props - The props for the component.
 * @returns {JSX.Element} The rendered Alert component.
 */
export const Alert: React.FC<AlertProps> = React.memo(({ type = 'info', message, className = '' }) => {
  const base = 'rounded-md p-4 flex items-center gap-2';
  const variants = {
    info: 'bg-blue-100 text-blue-800',
    success: 'bg-green-100 text-green-800',
    warning: 'bg-yellow-100 text-yellow-800',
    error: 'bg-red-100 text-red-800',
  };
  return (
    <div className={`${base} ${variants[type]} ${className}`} role="alert">
      {type === 'info' && <span aria-label="Info">ℹ️</span>}
      {type === 'success' && <span aria-label="Success">✅</span>}
      {type === 'warning' && <span aria-label="Warning">⚠️</span>}
      {type === 'error' && <span aria-label="Error">❌</span>}
      <span>{message}</span>
    </div>
  );
});

Alert.displayName = 'Alert';

Alert.propTypes = {
  type: PropTypes.oneOf(['info', 'success', 'warning', 'error']),
  message: PropTypes.string.isRequired,
  className: PropTypes.string,
};

export default Alert; 