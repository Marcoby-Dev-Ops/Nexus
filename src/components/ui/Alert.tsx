import React from 'react';
import { cn } from '@/lib/utils';
import PropTypes from 'prop-types';

/**
 * @interface AlertProps
 * @description Props for the Alert component.
 */
export interface AlertProps extends React.HTMLAttributes<HTMLDivElement> {
  variant?: 'default' | 'error' | 'success' | 'warning';
  action?: React.ReactNode;
}

/**
 * @interface AlertDescriptionProps
 * @description Props for the AlertDescription component.
 */
export interface AlertDescriptionProps extends React.HTMLAttributes<HTMLParagraphElement> {}

/**
 * @name Alert
 * @description A notification component for feedback messages.
 * @param {AlertProps} props - The props for the component.
 * @returns {JSX.Element} The rendered Alert component.
 */
export const Alert: React.FC<AlertProps> = ({
  children,
  className,
  variant = 'default',
  action,
  ...props
}) => {
  const variants = {
    default: 'bg-muted text-foreground',
    error: 'bg-destructive/10 text-destructive',
    success: 'bg-success/10 text-success',
    warning: 'bg-warning/10 text-warning'
  };

  return (
    <div
      className={cn(
        'rounded-lg p-4 flex items-start gap-4',
        variants[variant],
        className
      )}
      {...props}
    >
      <div className="flex-1">{children}</div>
      {action && <div className="flex-shrink-0">{action}</div>}
    </div>
  );
};

/**
 * @name AlertDescription
 * @description A description component for use within Alert components.
 * @param {AlertDescriptionProps} props - The props for the component.
 * @returns {JSX.Element} The rendered AlertDescription component.
 */
export const AlertDescription: React.FC<AlertDescriptionProps> = ({
  className,
  ...props
}) => (
  <div
    className={cn('text-sm [&_p]:leading-relaxed', className)}
    {...props}
  />
);

Alert.displayName = 'Alert';
AlertDescription.displayName = 'AlertDescription';

Alert.propTypes = {
  variant: PropTypes.oneOf(['default', 'error', 'success', 'warning']),
  action: PropTypes.node,
  className: PropTypes.string,
};

AlertDescription.propTypes = {
  className: PropTypes.string,
};

export default Alert; 