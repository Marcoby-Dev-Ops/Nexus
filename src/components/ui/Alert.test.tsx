/**
 * @file Alert.test.tsx
 * @description Unit and snapshot tests for the Alert component.
 */
import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Alert from './Alert';

describe('Alert', () => {
  it('renders with message', () => {
    render(<Alert message="Test Alert" />);
    expect(screen.getByText('Test Alert')).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Alert message="Snapshot Alert" />);
    expect(asFragment()).toMatchSnapshot();
  });
}); 