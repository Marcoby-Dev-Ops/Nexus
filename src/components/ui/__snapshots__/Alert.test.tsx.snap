// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="rounded-lg p-4 flex items-start gap-4 bg-muted text-foreground"
  >
    <div
      class="flex-1"
    >
      Snapshot Alert
    </div>
  </div>
</DocumentFragment>
`;
